go/libraries/doltcore/env/actions/reset.go

```diff
@@ -154,7 +154,17 @@ func ResetHard(
 		return err
 	}
 
```

Comment:
in instances like this, it would be nice to pass less repo state in to the library function. We could resolve the current working branch outside of the `actions` package and avoid passing `dEnv` altogether

Reply:
Yeah the actions package in general is kind of a mess, there's a ton of stuff there that's only called from CLI path. I'll do another pass to remove some of it.